File: Professional Image Modifier 4/[C#]-Professional Image Modifier 4/C#/ImageFunctions/Controls/SusanCornerProperties.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ImageFunctions.Controls
{
	public partial class SusanCornerProperties : UserControl
	{

		public int GeometricalDefault { get; set; } // As set by Aforge
		public int DifferenceDefault { get; set; }  // As set by Aforge
		public int GeometricalThreshold { get; set; } // Set by User
		public int DifferenceThreshold { get; set; } // Set by User

		public SusanCornerProperties()
		{
			InitializeComponent();
			GeometricalDefault = 18;
			DifferenceDefault = 25;
			trackDifferenceThreshold.Value = DifferenceDefault;
			trackGeometricalThreshold.Value = GeometricalDefault;
		}


		private void trackGeometricalThreshold_ValueChanged(object sender, EventArgs e)
		{
			lblGeometricalThreshold.Text = trackGeometricalThreshold.Value.ToString();
			GeometricalThreshold = trackGeometricalThreshold.Value;
		}

		private void trackDifferenceThreshold_ValueChanged(object sender, EventArgs e)
		{
			lblDifferenceThreshold.Text = trackDifferenceThreshold.Value.ToString();
			DifferenceThreshold = trackDifferenceThreshold.Value;
		}

		public void SetDefaults()
		{
			trackDifferenceThreshold.Value = DifferenceDefault;
			trackGeometricalThreshold.Value = GeometricalDefault;
		}
	}
}

File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Controls/SusanCornerProperties.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace ImageFunctions.Controls
	Partial Public Class SusanCornerProperties
		Inherits UserControl

		Public Property GeometricalDefault() As Integer
		Public Property DifferenceDefault() As Integer
		Public Property GeometricalThreshold() As Integer
		Public Property DifferenceThreshold() As Integer

		Public Sub New()
			InitializeComponent()
			GeometricalDefault = 18
			DifferenceDefault = 25
			trackDifferenceThreshold.Value = DifferenceDefault
			trackGeometricalThreshold.Value = GeometricalDefault
		End Sub


		Private Sub trackGeometricalThreshold_ValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trackGeometricalThreshold.ValueChanged
			lblGeometricalThreshold.Text = trackGeometricalThreshold.Value.ToString()
			GeometricalThreshold = trackGeometricalThreshold.Value
		End Sub

		Private Sub trackDifferenceThreshold_ValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trackDifferenceThreshold.ValueChanged
			lblDifferenceThreshold.Text = trackDifferenceThreshold.Value.ToString()
			DifferenceThreshold = trackDifferenceThreshold.Value
		End Sub

		Public Sub SetDefaults()
			trackDifferenceThreshold.Value = DifferenceDefault
			trackGeometricalThreshold.Value = GeometricalDefault
		End Sub
	End Class
End Namespace
